src/controllers/GetPosts.ts

```diff
@@ -0,0 +1,52 @@
+import { Response, Request } from 'express'
+import { User, Post, Like, Tag, Animal } from '../db'
+import { Op } from 'sequelize'
+
+const getPosts = async (req: Request, res: Response) => {
+  const { tagId, animalId, searchCountent } = req.query
+  console.log(req.query)
+
+  let filterData = { }
+  if (tagId && animalId) {
+    filterData = {
+      content: {
+        [Op.substring]: searchCountent || ''
+      },
+      TagId: tagId,
+      AnimalId: animalId
+    }
+  } else if (animalId) {
+    filterData = {
+      content: {
+        [Op.substring]: searchCountent || ''
+      },
+      AnimalId: animalId
+    }
+  } else if (tagId) {
+    filterData = {
+      content: {
+        [Op.substring]: searchCountent || ''
+      },
+      TagId: tagId
+    }
+  } else {
+    filterData = {
+      content: {
+        [Op.substring]: ''
```

Comment:
there is a bug here
if there is no `tagId` or `animalId` you put the search to empty string
that means get all the data even if there is something in the `searchContent` 

and there is no test case for this situation